src/eventPayloadSchemaBuilder.ts

```diff
@@ -0,0 +1,70 @@
+const yup = require('yup');
+
+export let schemaOnPush = yup.object().shape({
```

Comment:
Kind of sad that yup does not seem able to handle typescript unions as io-ts can because that would allow us to have a single step to validate the payload, whichever of the four types it is without having to switch and decide which validator to use. Effectively we would be able to use the type system to express the conditionality. It's a trade-off between complexity and function. I get that io-ts is a bit mind bending though so don't feel too strongly about it.

Reply:
I spent a lot of time trying to get io-ts to work and ended up with a non-elegant solution that I would really not happily maintain myself. It is definitely more powerful than yup (I also get the feeling it has been around for longer --> maybe I am wrong?) but I find yup easier to come back to if/when needed. My experience with io-ts was surely heavily biased by my (in)experience in typescript but when it comes to readability and ease of implementation for me yup wins  --> My personal opinion klaxon :D